src/Uno.UI/UI/Xaml/Media/Imaging/BitmapSource.cs

```diff
@@ -67,56 +67,63 @@ protected BitmapSource(string sourceString) : base(sourceString)
 		/// Helper for Uno... not part of UWP contract
 		/// </summary>
 		public void SetSource(Stream streamSource)
-		{
-			SetSource(streamSource.AsRandomAccessStream());
-		}
+			=> SetSourceCore(streamSource.AsRandomAccessStream());
 
 		/// <summary>
 		/// Helper for Uno... not part of UWP contract
 		/// </summary>
-		public async Task SetSourceAsync(Stream streamSource)
-			=> await SetSourceAsync(streamSource.AsRandomAccessStream());
+		public Task SetSourceAsync(Stream streamSource)
+		{
+			SetSourceCore(streamSource.AsRandomAccessStream());
+			return ForceLoad(CancellationToken.None);
+		}
 
 		public void SetSource(IRandomAccessStream streamSource)
-			// We prefer to use the SetSourceAsync here in order to make sure that the stream is copied ASYNChronously,
-			// which is important since we are using a stream wrapper of and <In|Out|RA>Stream which might freeze the UI thread / throw exception.
-			=> Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => SetSourceAsync(streamSource));
+			=> SetSourceCore(streamSource);
 
 		public IAsyncAction SetSourceAsync(IRandomAccessStream streamSource)
 		{
-			async Task SetSourceAsync(CancellationToken ct)
+			SetSourceCore(streamSource);
+			return AsyncAction.FromTask(ForceLoad);
+		}
+
+		private void SetSourceCore(IRandomAccessStream streamSource)
+		{
+			if (streamSource == null)
 			{
-				if (streamSource == null)
-				{
-					//Same behavior as windows, although the documentation does not mention it!!!
-					throw new ArgumentException(nameof(streamSource));
-				}
+				//Same behavior as windows, although the documentation does not mention it!!!
+				throw new ArgumentException(nameof(streamSource));
+			}
 
-				PixelWidth = 0;
-				PixelHeight = 0;
+			PixelWidth = 0;
+			PixelHeight = 0;
 
-#if __NETSTD__
-				_stream = streamSource.CloneStream();
+			// The source has to be cloned before leaving the "SetSource[Async]".
+			var clonedStreamSource = streamSource.CloneStream();
 
-				var tcs = new TaskCompletionSource<object>();
+#if __NETSTD__
+			_stream = clonedStreamSource;
+#else
+			Stream = clonedStreamSource.AsStream();
```

Comment:
Was this intended? the property instead of the field? `Stream` / `_stream`

Reply:
The `Stream` property is in the `ImageSource` base class and is present only on iOS and Android. On .net standard we have moved it back into a private field of the `BitmapSource` as it's this class which has the ability to load an image from a `Stream` (or actually an `IRandomAccessStream`)